src/modules/keyboardmanager/common/Shortcut.cpp

```diff
@@ -775,25 +775,18 @@ KeyboardManagerHelper::ErrorType Shortcut::DoKeysOverlap(const Shortcut& first,
-        else if (first.winKey != ModifierKey::Disabled && second.winKey != ModifierKey::Disabled && (first.winKey == ModifierKey::Both || second.winKey == ModifierKey::Both))
+        // action keys match
+        else if (first.actionKey == second.actionKey)
         {
-            return KeyboardManagerHelper::ErrorType::ConflictingModifierShortcut;
-        }
-        // If both have ctrl key modifiers and one is the both version then there will be an overlap
-        else if (first.ctrlKey != ModifierKey::Disabled && second.ctrlKey != ModifierKey::Disabled && (first.ctrlKey == ModifierKey::Both || second.ctrlKey == ModifierKey::Both))
-        {
-            return KeyboardManagerHelper::ErrorType::ConflictingModifierShortcut;
-        }
-        // If both have alt key modifiers and one is the both version then there will be an overlap
-        else if (first.altKey != ModifierKey::Disabled && second.altKey != ModifierKey::Disabled && (first.altKey == ModifierKey::Both || second.altKey == ModifierKey::Both))
-        {
-            return KeyboardManagerHelper::ErrorType::ConflictingModifierShortcut;
-        }
-        // If both have shift key modifiers and one is the both version then there will be an overlap
-        else if (first.shiftKey != ModifierKey::Disabled && second.shiftKey != ModifierKey::Disabled && (first.shiftKey == ModifierKey::Both || second.shiftKey == ModifierKey::Both))
-        {
-            return KeyboardManagerHelper::ErrorType::ConflictingModifierShortcut;
+            // corresponding modifiers are either both disabled or both not disabled - this ensures that both match in types of modifers i.e. Ctrl(l/r/c) Shift (l/r/c) A matches Ctrl(l/r/c) Shift (l/r/c) A
+            if (((first.winKey != ModifierKey::Disabled && second.winKey != ModifierKey::Disabled) || (first.winKey == ModifierKey::Disabled && second.winKey == ModifierKey::Disabled)) && ((first.ctrlKey != ModifierKey::Disabled && second.ctrlKey != ModifierKey::Disabled) || (first.ctrlKey == ModifierKey::Disabled && second.ctrlKey == ModifierKey::Disabled)) && ((first.altKey != ModifierKey::Disabled && second.altKey != ModifierKey::Disabled) || (first.altKey == ModifierKey::Disabled && second.altKey == ModifierKey::Disabled)) && ((first.shiftKey != ModifierKey::Disabled && second.shiftKey != ModifierKey::Disabled) || (first.shiftKey == ModifierKey::Disabled && second.shiftKey == ModifierKey::Disabled)))
+            {
+                // If one of the modifier is common
+                if ((first.winKey == ModifierKey::Both || second.winKey == ModifierKey::Both) || (first.ctrlKey == ModifierKey::Both || second.ctrlKey == ModifierKey::Both) || (first.altKey == ModifierKey::Both || second.altKey == ModifierKey::Both) || (first.shiftKey == ModifierKey::Both || second.shiftKey == ModifierKey::Both))
+                {
+                    return KeyboardManagerHelper::ErrorType::ConflictingModifierShortcut;
+                }
+            }
         }

```

Comment:
Maybe we should break these conditions into multiple lines so its easier to read

Reply:
Should I make them nested `if`s instead?